core/runtime/src/main/java/io/quarkus/runtime/PreShutdownEvent.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.runtime;
+
+/**
+ * Event payload that is fired in a pre-shutdown phase if {@code quarkus.shutdown.delay-enabled=true}.
+ * <p>
+ * This event is observed as follows:
+ *
+ * <code><pre>
+ *     void onPreShutdown(@Observes PreShutdownEvent ev) {
+ *         LOGGER.info("The application is about to shutdown...");
+ *     }
+ * </pre></code>
+ *
+ * The annotated method can access other injected beans.
+ */
+public class PreShutdownEvent {
```

Comment:
I wonder if we should have `@PreShutdown` as well, i.e. similarly to `io.quarkus.runtime.Shutdown`.